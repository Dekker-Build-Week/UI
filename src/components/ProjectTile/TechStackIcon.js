import React from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from "@material-ui/core/styles";
import techStackstyles from "assets/jss/material-dashboard-react/components/techStackStyle.js";

const useStyles = makeStyles(techStackstyles);

const TechStackIcon = (props) => {
    const classes = useStyles();

    const {techStack} = props;

    return (
        <li className = {classes.techList}>
            <img src = {techStack} alt = "tech" className = {classes.techStack} />
        </li>
    );
}

TechStackIcon.propTypes = {
    TechStack : PropTypes.object.isRequired
}

export default TechStackIcon;
