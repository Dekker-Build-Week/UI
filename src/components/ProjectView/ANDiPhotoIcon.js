import React from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from "@material-ui/core/styles";
import andiPhotoStyles from "assets/jss/material-dashboard-react/components/andiPhotoProjectStyle.js";

const useStyles = makeStyles(andiPhotoStyles);

const ANDiPhotoIcon = (props) => {
    const classes = useStyles();

    const { ANDiPhoto } = props;

    return (
        <li className = {classes.ANDiList}>
            <img src = {ANDiPhoto} alt = "ANDi" className = {classes.ANDiPhoto} />
        </li>
    );
}

ANDiPhotoIcon.propTypes = {
    ANDiPhoto : PropTypes.string.isRequired
}

export default ANDiPhotoIcon;
