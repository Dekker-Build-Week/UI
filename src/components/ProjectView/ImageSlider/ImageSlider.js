import React, { useState, useEffect } from 'react'
import { makeStyles } from "@material-ui/core/styles";
import imageSliderStyle from 'assets/jss/material-dashboard-react/components/imageSliderStyle.js';
import { useTransition, animated, config } from 'react-spring'

const useStyles = makeStyles(imageSliderStyle);

export default (props) => {
    const classes = useStyles();

    const [index, set] = useState(0)

    const { images } = props;      

    const transitions = useTransition(images[index], item => item.position, {
        from: { opacity: 0 },
        enter: { opacity: 1 },
        leave: { opacity: 0 },
        config: config.molasses,
    })

    useEffect(
        () => {
            const interval = setInterval(() => {
                // eslint-disable-next-line
                set(state => (state + 1) % (images.length));
            }, 4000);
            
            return () => {
                clearInterval(interval);
            }
        }, [])
  
    
    return (
        <div className = {classes.root}>
            {
                transitions.map(({ item, props, key }) => {
                    return (
                        <animated.div
                            key={key}
                            className= {classes.scriptBox}
                            style={{ ...props, backgroundImage: `url(${item.source})` }}
                        />
                    );
                })
            }
        </div>
    )
}
