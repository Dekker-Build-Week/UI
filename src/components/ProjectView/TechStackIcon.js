import React from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from "@material-ui/core/styles";
import techStackstyles from "assets/jss/material-dashboard-react/components/techStackProjectStyle.js";

const useStyles = makeStyles(techStackstyles);

const TechStackIcon = (props) => {
    const classes = useStyles();

    const {techStack} = props;

    return (
        <li className = {classes.techList}>
            <img src = {techStack} alt = "tech" className = {classes.techStack} />
        </li>
    );
}

TechStackIcon.propTypes = {
    techStack : PropTypes.string.isRequired
}

export default TechStackIcon;
