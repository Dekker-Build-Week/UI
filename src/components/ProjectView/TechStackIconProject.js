import React from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from "@material-ui/core/styles";
import techStackstyles from "assets/jss/material-dashboard-react/components/techStackProjectStyle.js";

const useStyles = makeStyles(techStackstyles);

const TechStackProjectIcon = (props) => {
    const classes = useStyles();

    const {techStack} = props;

    return (
        <li className = {classes.techList}>
            <img src = {techStack} alt = "tech" className = {classes.techStack} />
        </li>
    );
}

TechStackProjectIcon.propTypes = {
    techStack : PropTypes.string.isRequired
}

export default TechStackProjectIcon;
