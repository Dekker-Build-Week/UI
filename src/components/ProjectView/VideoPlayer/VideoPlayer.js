import React from 'react';
import { makeStyles } from "@material-ui/core/styles";
import videoPlayerStyle from 'assets/jss/material-dashboard-react/components/videoPlayerStyle.js';
import { Player } from 'video-react';

const useStyles = makeStyles(videoPlayerStyle);

export default (props) => {
    const classes = useStyles();

    const { videoSource } = props;

    return (
        <div className = {classes.videoContainer}>
            <Player autoPlay className = {classes.videoPlayer} >
                <source src = { videoSource ? videoSource : "https://media.w3.org/2010/05/sintel/trailer_hd.mp4" } />
            </Player>
        </div>
    );
}