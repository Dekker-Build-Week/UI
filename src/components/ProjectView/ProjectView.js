import React from "react";
import { makeStyles } from "@material-ui/core/styles";

import Card from "components/Card/Card.js";
import CardHeader from "components/Card/CardHeader.js";
import CardFooter from "components/Card/CardFooter.js";
import CardBody from "components/Card/CardBody";
import projectViewStyle from "assets/jss/material-dashboard-react/components/projectViewStyle.js";
import GridItem from "components/Grid/GridItem";
import GridContainer from "components/Grid/GridContainer";
import TechStackIcon from "./TechStackIcon";
import ANDiPhotoIcon from "./ANDiPhotoIcon";
import ImageSlider from "./ImageSlider/ImageSlider";
import ReactPlayer from "react-player";

const useStyles = makeStyles(projectViewStyle);

const ProjectView = props => {
  const classes = useStyles();

  const {
    projectTitle,
    clientName,
    projectDescription,
    clientLogo,
    coverImage,
    imagePaths,
    team,
    techStack,
    clientVideo
  } = props;
  return (
    <div className={classes.projectView}>
      <Card>
        <CardHeader stats icon className={classes.cardHeader}>
          <img
            src={imagePaths + "/" + clientLogo}
            alt="Client Logo"
            className={classes.clientLogo}
          />
          <p className={classes.clientName}>{clientName}</p>
          <h3 className={classes.projectTitle}>{projectTitle}</h3>
          <div className={classes.projectBorderTop}></div>
        </CardHeader>
        <CardBody className={classes.projectBody}>
          <div className={classes.projectInfo}>
            <GridContainer>
              <GridItem xs={12} sm={12} md={8}>
                <div className={classes.contentContainer}>
                  <ReactPlayer url={imagePaths + "/" + clientVideo} playing />
                </div>
              </GridItem>
              <GridItem xs={12} sm={12} md={4}>
                <GridContainer>
                  <GridItem xs={12} sm={12} md={12}>
                    <div className={classes.projectDescriptionContainer}>
                      <p className={classes.projectDescription}>
                        {projectDescription}
                      </p>
                    </div>
                  </GridItem>
                  <GridItem xs={12} sm={12} md={12}>
                    <div className={classes.teamIcons}>
                      <br />
                      <h6>Team Members</h6>
                      <GridContainer>
                        {team.map((andi, index) => {
                          return (
                            <ANDiPhotoIcon
                              key={index}
                              ANDiPhoto={imagePaths + "/" + andi.photoPath}
                              ANDiName={andi.name}
                            />
                          );
                        })}
                      </GridContainer>
                    </div>
                  </GridItem>
                </GridContainer>
              </GridItem>
            </GridContainer>
          </div>
        </CardBody>
        <CardFooter className={classes.projectFooter}>
          <ul className={classes.projectANDis}>
            {team
              ? techStack.map((tech, index) => {
                  return (
                    <TechStackIcon
                      key={index}
                      techStack={imagePaths + "/" + tech.imagePath}
                    />
                  );
                })
              : null}
          </ul>
        </CardFooter>
      </Card>
    </div>
  );
};

export default ProjectView;
