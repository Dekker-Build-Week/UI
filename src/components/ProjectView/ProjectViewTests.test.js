import React from 'react';

import ANDiPhotoIcon from './ANDiPhotoIcon';
import TechStackIcon from './TechStackIcon';
import ProjectView from './ProjectView';
import ImageSlider from './ImageSlider/ImageSlider';
import VideoPlayer from './VideoPlayer/VideoPlayer';
import renderer from 'react-test-renderer';

describe('ProjectView test suite', () => {
    it('Renders the ProjectView correctly', () => {
        const tree = renderer
            .create(<ProjectView></ProjectView>)
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    }),
    it('Renders the ANDiPhotoIcon correctly', () => {
        const tree = renderer
            .create(<ANDiPhotoIcon
                ANDiPhoto = "Photo"></ANDiPhotoIcon>)
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    }),
    it('Renders a TechStackIcon with GridItems correctly', () => {
        const tree = renderer 
            .create(<TechStackIcon
                        techStack = "Stack"></TechStackIcon>)
                .toJSON();

        expect(tree).toMatchSnapshot(); 
    }),
    it('Renders the ImageSlider successfully', () => {
        const tree = renderer 
            .create(<ImageSlider
                        images = {[]}></ImageSlider>)
                .toJSON();

        expect(tree).toMatchSnapshot(); 
    })
    ,
    it('Renders the VideoPlayer successfully', () => {
        const tree = renderer 
            .create(<VideoPlayer
                        videoSource = ""></VideoPlayer>)
                .toJSON();

        expect(tree).toMatchSnapshot(); 
    })
});