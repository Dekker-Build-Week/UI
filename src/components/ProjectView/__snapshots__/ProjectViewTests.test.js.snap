// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectView test suite Renders a TechStackIcon with GridItems correctly 1`] = `
<li
  className="makeStyles-techList-145"
>
  <img
    alt="tech"
    className="makeStyles-techStack-144"
    src="Stack"
  />
</li>
`;

exports[`ProjectView test suite Renders the ANDiPhotoIcon correctly 1`] = `
<li
  className="makeStyles-ANDiList-143"
>
  <img
    alt="ANDi"
    className="makeStyles-ANDiPhoto-142"
    src="Photo"
  />
</li>
`;

exports[`ProjectView test suite Renders the ImageSlider successfully 1`] = `
<div
  className="makeStyles-root-136"
/>
`;

exports[`ProjectView test suite Renders the ProjectView correctly 1`] = `
<div
  className="makeStyles-projectView-1"
>
  <div
    className="makeStyles-card-13"
  >
    <div
      className="makeStyles-cardHeader-17 makeStyles-cardHeaderStats-19 makeStyles-cardHeaderIcon-20"
    >
      <img
        alt="Client Logo"
        className="makeStyles-clientLogo-2"
        src="https://undefined"
      />
      <p
        className="makeStyles-clientName-3"
      />
      <h3
        className="makeStyles-projectTitle-4"
      />
      <div
        className="makeStyles-projectBorderTop-9"
      />
    </div>
    <div
      className="makeStyles-cardBody-27 makeStyles-projectBody-5"
    >
      <div
        className="makeStyles-projectInfo-6"
      >
        <div
          className="MuiGrid-root makeStyles-grid-30 MuiGrid-container"
        >
          <div
            className="MuiGrid-root makeStyles-grid-134 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-8"
          >
            <div
              className="makeStyles-contentContainer-7"
            >
              <div
                className="makeStyles-root-136"
              />
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-grid-134 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4"
          >
            <p
              className="makeStyles-projectDescription-8"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="makeStyles-cardFooter-137 makeStyles-projectFooter-10"
    >
      <ul
        className="makeStyles-projectANDis-11"
      />
      <ul
        className="makeStyles-projectTechstacks-12"
      />
    </div>
  </div>
</div>
`;

exports[`ProjectView test suite Renders the VideoPlayer successfully 1`] = `
<div
  className="makeStyles-videoContainer-147"
>
  <div
    className="video-react-controls-enabled video-react-paused video-react-fluid video-react-user-active video-react-workinghover video-react makeStyles-videoPlayer-146"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="region"
    style={
      Object {
        "paddingTop": "56.25%",
      }
    }
    tabIndex="-1"
  >
    <video
      autoPlay={true}
      className="video-react-video"
      muted={false}
      onAbort={[Function]}
      onCanPlay={[Function]}
      onCanPlayThrough={[Function]}
      onDurationChange={[Function]}
      onEmptied={[Function]}
      onEnded={[Function]}
      onError={[Function]}
      onLoadStart={[Function]}
      onLoadedData={[Function]}
      onLoadedMetadata={[Function]}
      onPause={[Function]}
      onPlay={[Function]}
      onPlaying={[Function]}
      onProgress={[Function]}
      onRateChange={[Function]}
      onSeeked={[Function]}
      onSeeking={[Function]}
      onStalled={[Function]}
      onSuspend={[Function]}
      onTimeUpdate={[Function]}
      onVolumeChange={[Function]}
      onWaiting={[Function]}
      playsInline={false}
      preload="auto"
      tabIndex="-1"
    />
    <div
      className="video-react-loading-spinner"
    />
    <button
      aria-live="polite"
      className="video-react-big-play-button video-react-big-play-button-left big-play-button-hide"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="video-react-control-text"
      >
        Play Video
      </span>
    </button>
    <div
      className="video-react-control-bar video-react-control-bar-auto-hide"
    >
      <button
        className="video-react-play-control video-react-control video-react-button video-react-paused"
        onClick={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="video-react-control-text"
        >
          Play
        </span>
      </button>
      <div
        className="video-react-volume-menu-button-horizontal video-react-vol-3 video-react-volume-menu-button video-react-menu-button-inline video-react-control video-react-button video-react-menu-button"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="video-react-menu"
          onClick={[Function]}
        >
          <div
            className="video-react-menu-content"
          >
            <div
              aria-label="volume level"
              aria-valuemax={100}
              aria-valuemin={0}
              aria-valuenow="100.00"
              aria-valuetext="100.00%"
              className="video-react-volume-bar video-react-slider-bar video-react-slider-horizontal video-react-slider"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              role="slider"
              tabIndex="0"
            >
              <div
                className="video-react-volume-level"
                style={
                  Object {
                    "width": "100.00%",
                  }
                }
              >
                <span
                  className="video-react-control-text"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="video-react-current-time video-react-time-control video-react-control"
      >
        <div
          aria-live="off"
          className="video-react-current-time-display"
        >
          <span
            className="video-react-control-text"
          >
            Current Time 
          </span>
          0:00
        </div>
      </div>
      <div
        className="video-react-time-control video-react-time-divider"
        dir="ltr"
      >
        <div>
          <span>
            /
          </span>
        </div>
      </div>
      <div
        className="video-react-duration video-react-time-control video-react-control"
      >
        <div
          aria-live="off"
          className="video-react-duration-display"
        >
          <span
            className="video-react-control-text"
          >
            Duration Time 
          </span>
          0:00
        </div>
      </div>
      <div
        className="video-react-progress-control video-react-control"
        onMouseMove={[Function]}
      >
        <div
          aria-label="video progress bar"
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow="NaN"
          aria-valuetext="0:00"
          className="video-react-progress-holder video-react-slider-horizontal video-react-slider"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex="0"
        >
          <div
            className="video-react-play-progress video-react-slider-bar"
            data-current-time="0:00"
            style={
              Object {
                "width": "NaN%",
              }
            }
          >
            <span
              className="video-react-control-text"
            >
              Progress: NaN%
            </span>
          </div>
        </div>
      </div>
      <button
        className="video-react-icon-fullscreen video-react-fullscreen-control video-react-control video-react-button video-react-icon"
        onClick={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="video-react-control-text"
        >
          Non-Fullscreen
        </span>
      </button>
    </div>
  </div>
</div>
`;
